### PR TITLE
Ctw 282/add meatball menu to conditions

### PR DESCRIPTION
**Pull request by luthfib**
Adds dropdown menu and ellipses action button and provisionally opens the same drawer for now. Next stories that create those drawers should handle open the appropriate drawers. 

**Potential fixes**: Centering the menu and arrow to go in the middle of the ellipses button

image

### CODE REVIEW

File: src/components/core/table/table-rows.tsx [hunk 1]
```diff
@@ -62,17 +68,34 @@ export const TableRows = <T extends MinRecordItem>({
...
-            <TableDataCell
-              key={column.title ?? index}
-              column={column}
-              record={record}
-              index={index}
-              showLeftTableBorderShadow={showLeftTableBorderShadow}
-            />
-          ))}
-        </tr>
+        <Fragment key={record.id}>
```
Comment (WesleyKapow) [inline src/components/core/table/table-rows.tsx:71]: Fragment not needed. Can continue using `key` on the `tr` as before.

File: src/components/core/table/table-rows.tsx [hunk 2]
```diff
@@ -62,17 +68,34 @@ export const TableRows = <T extends MinRecordItem>({
...
+                className={cx(
+                  "ctw-table-action-column  ctw-min-w-[4rem] ctw-text-center ctw-text-icon-default",
+                  {
+                    "ctw-table-action-column-sticky":
+                      showRightTableBorderShadow,
+                  }
+                )}
+              >
+                <DropdownMenu menuItems={rowActions}>
+                  <DotsHorizontalIcon className="ctw-2-5 ctw-w-5 ctw-cursor-pointer" />
```
Comment (WesleyKapow) [inline src/components/core/table/table-rows.tsx:93]: What's `ctw-2-5`?

File: src/components/core/dropdown-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            <Float.Arrow className="ctw-absolute ctw-h-5 ctw-w-5 ctw-rotate-45 ctw-border ctw-border-black ctw-bg-white" />
+            <div className="ctw-relative ctw-overflow-hidden ctw-bg-white">
+              {menuItems.map((menuItem) => (
+                <Menu.Item key={menuItem.name}>
+                  {({ active }) => (
+                    <button
+                      onClick={() => menuItem.action()}
+                      type="button"
+                      className={cx(
+                        "group ctw-flex ctw-w-full ctw-items-center ctw-border-none ctw-bg-transparent ctw-py-2 ctw-px-2 ctw-text-content-black",
```
Comment (WesleyKapow) [inline src/components/core/dropdown-menu.tsx:43]: Should be `ctw-group`? Maybe not even needed.

**Review by WesleyKapow — CHANGES_REQUESTED** [review]

File: src/components/core/table/table.tsx [hunk 4]
```diff
@@ -91,7 +92,8 @@ export const Table = <T extends MinRecordItem>({
         className={cx(
           "ctw-relative ctw-overflow-hidden ctw-rounded-lg ctw-p-px",
           {
-            "ctw-table-scroll-right-shadow": showRightTableBorderShadow,
+            "ctw-table-scroll-right-shadow":
+              ellipsesMenuAction && showRightTableBorderShadow,
```
[the remark this answers is not in the document]
Comment (luthfib) [reply src/components/core/table/table.tsx]: Ahh good catch. It should actually be !ellipsesMenuAction && showRightTableBorderShadow. The idea being that overall table shadow on the right should only show if ellipsesMenuAction is not defined.

File: src/components/content/conditions.tsx [hunk 5]
```diff
@@ -144,13 +144,6 @@ export function Conditions({ className, patientUPID }: ConditionsProps) {
               isLoading={notReviewedConditionsIsLoading}
               showTableHead={false}
               message={notReviewedConditionsMessage}
-              rowActions={[
-                { name: "Add", action: () => setAddConditionIsOpen(true) },
-                {
-                  name: "View History",
-                  action: () => setAddConditionIsOpen(true),
-                },
-              ]}
```
Comment (WesleyKapow) [inline src/components/content/conditions.tsx]: This table will definitely have rowActions.. also important to keep the columns the same between the two tables for alignment purposes.

File: src/components/core/dropdown-menu.tsx [hunk 6]
```diff
@@ -16,45 +16,46 @@ export type DropdownMenuProps = {
...
+    <Menu>
+      <Float
+        portal
+        placement="bottom-start"
+        offset={10}
+        flip
+        arrow
+        tailwindcssOriginClass
+      >
+        <Menu.Button className="ctw-cursor-pointer ctw-border-none ctw-bg-transparent">
```
Comment (WesleyKapow) [inline src/components/core/dropdown-menu.tsx:28]: Use `ctw-btn-clear` instead.

File: .changeset/perfect-points-own.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": patch
+---
+
+Remove unused conditions table export
```
Comment (WesleyKapow) [inline .changeset/perfect-points-own.md:5]: This doesn't capture the changes here.

**Review by WesleyKapow — CHANGES_REQUESTED** [review]

**Review by WesleyKapow — CHANGES_REQUESTED** [review]

File: src/components/core/dropdown-menu.tsx [hunk 8]
```diff
@@ -25,7 +25,7 @@ export function DropdownMenu({ children, menuItems }: DropdownMenuProps) {
         arrow
         tailwindcssOriginClass
       >
-        <Menu.Button className="ctw-btn-clear ctw-cursor-pointer ctw-border-none">
+        <Menu.Button className="ctw-btn-clear ctw-cursor-pointer ctw-border-none ctw-text-primary-dark hover:ctw-text-primary-main">
```
Comment (WesleyKapow) [inline src/components/core/dropdown-menu.tsx:28]: `"ctw-btn-clear ctw-link"` is all you need.

Comment (luthfib) [reply src/components/core/dropdown-menu.tsx]: Not sure why its showing the diff wrong but I have `ctw-btn-clear ctw-link` now only here. Shows up correctly in the code changes tab

**Review by WesleyKapow — APPROVED** [review]
LGTM

**Pull request merged**